### PR TITLE
urn: monitor count of unregistered pids by days

### PR DESCRIPTION
**Pull request by BadrAly**
A new api to count documents with unregistered pids
with an optional parameters `days`.

* Completes urn config for two organisations.
* 
Co-Authored-by: Aly Badr <aly.badr@rero.ch>

### CODE REVIEW

File: sonar/config_sonar.py [hunk 1]
```diff
@@ -602,6 +602,11 @@
             'types': ['coar:c_db06'],
             'code': 6,
             'namespace': 'urn:nbn:ch:rero'
+        },
+        'usi': {
```
Comment (jma) [inline sonar/config_sonar.py:606]: Can you mention this in your commit message?

File: sonar/modules/documents/urn.py [hunk 2]
```diff
@@ -97,3 +100,25 @@ def create_urn(cls, record):
...
+        :param status: PID status by default N.
+        :param days: Number of days passed since the creation of the document.
+        :returns: Documents count.
+        """
+        count = 0
+        query = PersistentIdentifier.query\
+                .filter_by(pid_type=cls.urn_pid_type)\
+                .filter_by(status=status)
+        if uuuids := [str(uuid.object_uuid) for uuid in query.all()]:
+            from sonar.modules.documents.api import DocumentSearch
```
Comment (jma) [inline sonar/modules/documents/urn.py:117]: Please put this on the top of the file if it is possible.

File: sonar/modules/documents/urn.py [hunk 3]
```diff
@@ -97,3 +100,25 @@ def create_urn(cls, record):
                 except PIDAlreadyExists:
                     current_app.logger.error(
                         f'generated urn already exist for document: {doc_pid}')
+
+    @classmethod
+    def get_urn_pids(cls, status=PIDStatus.NEW, days=None):
+        """Get URN pids by status.
```
Comment (jma) [inline sonar/modules/documents/urn.py:106]: This comment does not correspond to the return value.

File: tests/api/monitoring/test_monitoring_views.py [hunk 4]
```diff
@@ -16,11 +16,41 @@
...
+import pytest
 from flask import url_for
 from invenio_accounts.testutils import login_user_via_session
 from invenio_db import db
 
+from sonar.modules.documents.api import DocumentRecord
+
+
+@pytest.fixture()
+def minimal_document(db, bucket_location, organisation):
```
Comment (jma) [inline tests/api/monitoring/test_monitoring_views.py:28]: minimal_thesis_document?

File: tests/api/monitoring/test_monitoring_views.py [hunk 5]
```diff
@@ -16,11 +16,41 @@
...
-
+import pytest
 from flask import url_for
 from invenio_accounts.testutils import login_user_via_session
 from invenio_db import db
 
+from sonar.modules.documents.api import DocumentRecord
+
+
+@pytest.fixture()
```
Comment (jma) [inline tests/api/monitoring/test_monitoring_views.py:27]: You can put this in a separate conftest.py file.

File: tests/api/monitoring/test_monitoring_views.py [hunk 6]
```diff
@@ -16,11 +16,41 @@
...
+def minimal_document(db, bucket_location, organisation):
+    record = DocumentRecord.create(
```
Comment (jma) [inline tests/api/monitoring/test_monitoring_views.py:29]: You can use the `make_document` fixture function.

**Review by jma — APPROVED** [review]

File: sonar/monitoring/views/__init__.py [hunk 7]
```diff
@@ -110,3 +111,19 @@ def elastic_search():
...
+
+
+@api_blueprint.route('/urn')
+def unregistered_urn():
+    """Count of unregistered urn pids.
+
+    :return: jsonified count information.
+    """
+    try:
+        days= 0
```
Comment (rerowep) [inline sonar/monitoring/views/__init__.py:123]: Space missing.

File: sonar/monitoring/views/__init__.py [hunk 8]
```diff
@@ -110,3 +111,19 @@ def elastic_search():
...
+        if args := request.args:
+            days = int(args.get('days'))
```
Comment (rerowep) [inline sonar/monitoring/views/__init__.py:125]: Maybe we need test for correct args or default value for get?

**Pull request merged**